properties/basic.py

```diff
@@ -93,6 +115,44 @@ def fget(self):
 
         return property(fget=fget, doc=scope.help)
 
+    def serialize(self, value):
+        """Serialize the property value to JSON
+
+        If no serializer has been registered, this uses to_json
+        """
+        if self.serializer:
+            return self.serializer(value)
+        if value is None:
+            return None
```

Comment:
I think shortcutting `None` is the right choice for the default behaviour. It is still relatively simple for someone to add more interesting behaviour if they want. Let me try to explain and see if you agree...

Properties have a `required` attribute. When they are part of a `HasProperties` class, HasProperties validation will fail unless all `required` properties are set. However, if a property is not `required` and has not been set, its value will be `None` and validation will pass. Therefore properties with `None` values simply haven't been set.

If I am serializing a `HasProperties` instance, I'd expect the properties I did not set to just not show up, rather than getting serialized as `None` (or `'null'`). Catching `value is None` before converting to JSON does this.

You are correct, though, that someone might want to do something more interesting with `None`s. In that case, they can give the `Property` a custom `serializer` function - this will always be called before `None`s are bypassed. Alternatively, someone could define their own `Property` subclass and override the `serialize` method to do something totally different.